server/channels/store/searchlayer/layer.go

```diff
@@ -111,6 +111,28 @@ func (s *SearchStore) indexUser(rctx request.CTX, user *model.User) {
 	}
 }
 
+func (s *SearchStore) indexChannelsForTeam(rctx request.CTX, teamMember *model.TeamMember) {
+	channels, err := s.Channel().GetPublicChannelsForTeam(teamMember.TeamId, 0, 100)
+	if err != nil {
+		rctx.Logger().Warn("Encountered error while indexing public channels", mlog.String("team_member_id", teamMember.UserId), mlog.Err(err))
+		return
+	}
+
+	for _, engine := range s.searchEngine.GetActiveEngines() {
+		if engine.IsIndexingEnabled() {
+			runIndexFn(rctx, engine, func(engineCopy searchengine.SearchEngineInterface) {
+				for _, channel := range channels {
+					if err := engineCopy.IndexChannel(rctx, channel, []string{teamMember.UserId}, []string{teamMember.UserId}); err != nil {
+						rctx.Logger().Warn("Encountered error indexing channel", mlog.String("channel_id", channel.Id), mlog.String("search_engine", engineCopy.GetName()), mlog.Err(err))
+						return
+					}
+					rctx.Logger().Debug("Indexed channel in search engine", mlog.String("search_engine", engineCopy.GetName()), mlog.String("channel_id", channel.Id))
+				}
+			})
+		}
+	}
+}
```

Comment:
This does not look right. We need to actually get all team members for each channel, and then update the channel. We need to call `(c *SearchChannelStore) indexChannel` instead.

```go
for _, channel := range channels {
	s.channel.indexChannel(rctx, channel)
}
```

Reply:
TIL, so we always in theory purge other users when we re-index?

Reply:
It's only channels. Yes, there is no other way because of the JSON structure and ES being a document database. The channel object contains the list of all team_members, so if a new member gets added to a team, all channel objects need to be updated.